### PR TITLE
add $ref guidance to schema docs

### LINKED ISSUE #972: Update docs with how to use schema `$ref`s
We should add generic documentation to `/docs/resource-schemas.md` for how `$ref`s work. Examples being:

1. Local schema reference
2. Referencing a schema in the same repository via relative path
3. Referencing any schema via absolute path

Although this is initially just needed for design kits, in phase II of the technical design for schema references, it would get used for libraries and assets too.

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #972

### CODE REVIEW

File: services/web-app/pages/contributing/schema.mdx [hunk 1]
```diff
@@ -535,3 +539,83 @@ For the value of the `navData` array, you can set the following keys.
...
+
+```yml path="Example carbon.yml"
+designKits:
+  sample-design-kit:
+  $ref: 'https://unpkg.com/@carbon-platform/resources/carbon.yml#/designKits/carbon-g10-sketch'
+```
+
+### Relative references
+
+A relative path to a resourced contained in a yml file within the current repo, e.g.:
```
Comment (andreancardona) [inline services/web-app/pages/contributing/schema.mdx:590]:
```suggestion
A relative path to a resource contained in a yml file within the current repo, e.g.:
```

**Review by andreancardona — APPROVED** [review]
approved with super small change! 🔥

File: services/web-app/pages/contributing/schema.mdx [hunk 2]
```diff
@@ -535,3 +539,83 @@ For the value of the `navData` array, you can set the following keys.
...
+
+</AnchorLinks>
+
+For certain resource keys, platform allows the object to be defined as a reference to another
+existing resource instead of writing the actual value:
+
+```yml path="Example carbon.yml"
+designKits:
+  sample-design-kit:
+  $ref: './design-kits.yml#/designKits/sample-design-kit'
```
Comment (mattrosno) [inline services/web-app/pages/contributing/schema.mdx:559]: Does this line need indented?

File: services/web-app/pages/contributing/schema.mdx [hunk 3]
```diff
@@ -535,3 +539,83 @@ For the value of the `navData` array, you can set the following keys.
...
+  sample-design-kit:
+    name: Sample Design Kit
+    tool: sketch
+    type: ui
+    status: stable
+    url: sketch://something
+    action: link
+```
+
+There are 3 ways you can define references in carbon: absolute, relative and local references.
```
Comment (mattrosno) [inline services/web-app/pages/contributing/schema.mdx:575]:
```suggestion
There are 3 ways you can define references in Carbon: absolute, relative and local references.
```

File: services/web-app/pages/contributing/schema.mdx (hunk 2 above)
Comment (mattrosno) [reply services/web-app/pages/contributing/schema.mdx:559]: Below examples as well?

File: services/web-app/pages/contributing/schema.mdx [hunk 4]
```diff
@@ -535,3 +539,83 @@ For the value of the `navData` array, you can set the following keys.
...
+
+### Absolute references
+
+An absolute url that points to a resource contained in a yml file , e.g.:
```
Comment (mattrosno) [inline services/web-app/pages/contributing/schema.mdx:579]:
```suggestion
An absolute URL that points to a resource contained in a yml file , e.g.:
```

File: services/web-app/pages/contributing/schema.mdx [hunk 5]
```diff
@@ -535,3 +539,83 @@ For the value of the `navData` array, you can set the following keys.
...
+```yml path="Example carbon.yml"
+designKits:
+  sample-design-kit:
+  $ref: 'https://unpkg.com/@carbon-platform/resources/carbon.yml#/designKits/carbon-g10-sketch'
+```
+
+### Relative references
+
+A relative path to a resource contained in a yml file within the current repo, e.g.:
+'./design-kits.yml#/designKits/sample-design-kit'
```
Comment (mattrosno) [inline services/web-app/pages/contributing/schema.mdx:591]: Should we use backticks instead of single quotes for these example values?

**Review by mattrosno — CHANGES_REQUESTED** [review]

File: services/web-app/pages/contributing/schema.mdx (hunk 2 above)
Comment (francinelucca) [reply services/web-app/pages/contributing/schema.mdx]: Done!

File: services/web-app/pages/contributing/schema.mdx (hunk 5 above)
Comment (francinelucca) [reply services/web-app/pages/contributing/schema.mdx]: Done!

**Pull request merged**